### ISSUE TITLE
Aircrack-ng "setuid" errors during compilation on Ubuntu 64-bit

### ISSUE THREAD

**Issue by aircrack-ng** [aircrack-ng/aircrack-ng#1299]
**Reported by Jano on 3 Apr 2014 16:35 UTC**

Hi to all:

```
jano@ghostLab:~/aircrack-ng$ lsb_release -a;uname -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 13.10
Release:	13.10
Codename:	saucy
Linux ghostLab 3.11.0-19-generic #33-Ubuntu SMP Tue Mar 11 18:48:34 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux
```
```
jano@ghostLab:~/aircrack-ng$ make sqlite=true unstable=true libnl=true pcre=true
make -C src all
make[ingresso nella directory "/home/jano/aircrack-ng/src"
gcc -g -W -Wall -O3 -D_FILE_OFFSET_BITS=64 -D_REVISION=2389  -I/usr/local/include -DHAVE_SQLITE  -DHAVE_PCRE -DCONFIG_LIBNL30 -I/usr/include/libnl3   -Wno-unused-but-set-variable -Wno-array-bounds -Iinclude   -c -o airbase-ng.o airbase-ng.c
airbase-ng.c: In function main:
airbase-ng.c:4566:11: warning: ignoring return value of setuid, declared with attribute warn_unused_result [-Wunused-result](1]:)
     setuid( getuid() );
           ^
gcc -g -W -Wall -O3 -D_FILE_OFFSET_BITS=64 -D_REVISION=2389  -I/usr/local/include -DHAVE_SQLITE  -DHAVE_PCRE -DCONFIG_LIBNL30 -I/usr/include/libnl3   -Wno-unused-but-set-variable -Wno-array-bounds -Iinclude  airbase-ng.o common.o crypto.o osdep/radiotap/radiotap.o -o airbase-ng -Losdep -losdep -lnl-genl-3 -lnl-3   -lssl -lcrypto  -lpthread
gcc -g -W -Wall -O3 -D_FILE_OFFSET_BITS=64 -D_REVISION=2389  -I/usr/local/include -DHAVE_SQLITE  -DHAVE_PCRE -DCONFIG_LIBNL30 -I/usr/include/libnl3   -Wno-unused-but-set-variable -Wno-array-bounds -Iinclude   -c -o tkiptun-ng.o tkiptun-ng.c
tkiptun-ng.c: In function main:
tkiptun-ng.c:4195:11: warning: ignoring return value of setuid, declared with attribute warn_unused_result [    setuid( getuid() );
           ^
gcc -g -W -Wall -O3 -D_FILE_OFFSET_BITS=64 -D_REVISION=2389  -I/usr/local/include -DHAVE_SQLITE  -DHAVE_PCRE -DCONFIG_LIBNL30 -I/usr/include/libnl3   -Wno-unused-but-set-variable -Wno-array-bounds -Iinclude  tkiptun-ng.o common.o crypto.o osdep/radiotap/radiotap.o -o tkiptun-ng -Losdep -losdep -lnl-genl-3 -lnl-3   -lssl -lcrypto 
make[1](-Wunused-result]
): uscita dalla directory "/home/jano/aircrack-ng/src"
```

```
jano@ghostLab:~$ gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/4.8/lto-wrapper
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu/Linaro 4.8.1-10ubuntu9' --with-bugurl=file:///usr/share/doc/gcc-4.8/README.Bugs --enable-languages=c,c++,java,go,d,fortran,objc,obj-c++ --prefix=/usr --program-suffix=-4.8 --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --with-gxx-include-dir=/usr/include/c++/4.8 --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --enable-gnu-unique-object --enable-plugin --with-system-zlib --disable-browser-plugin --enable-java-awt=gtk --enable-gtk-cairo --with-java-home=/usr/lib/jvm/java-1.5.0-gcj-4.8-amd64/jre --enable-java-home --with-jvm-root-dir=/usr/lib/jvm/java-1.5.0-gcj-4.8-amd64 --with-jvm-jar-dir=/usr/lib/jvm-exports/java-1.5.0-gcj-4.8-amd64 --with-arch-directory=amd64 --with-ecj-jar=/usr/share/java/eclipse-ecj.jar --enable-objc-gc --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --with-tune=generic --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 4.8.1 (Ubuntu/Linaro 4.8.1-10ubuntu9)
```
 
----------------------------------------------

'''NOTE''':
- With Ubuntu Precise 32-bit I don't have this error message:
```
jano:~$ lsb_release -a; uname -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 12.04.4 LTS
Release:	12.04
Codename:	precise
Linux ghost 3.11.0-19-generic #33~precise1-Ubuntu SMP Wed Mar 12 21:17:09 UTC 2014 i686 i686 i386 GNU/Linux
```
```
jano:~$ gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/i686-linux-gnu/4.6/lto-wrapper
Target: i686-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu/Linaro 4.6.3-1ubuntu5' --with-bugurl=file:///usr/share/doc/gcc-4.6/README.Bugs --enable-languages=c,c++,fortran,objc,obj-c++ --prefix=/usr --program-suffix=-4.6 --enable-shared --enable-linker-build-id --with-system-zlib --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --with-gxx-include-dir=/usr/include/c++/4.6 --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --enable-gnu-unique-object --enable-plugin --enable-objc-gc --enable-targets=all --disable-werror --with-arch-32=i686 --with-tune=generic --enable-checking=release --build=i686-linux-gnu --host=i686-linux-gnu --target=i686-linux-gnu
Thread model: posix
gcc version 4.6.3 (Ubuntu/Linaro 4.6.3-1ubuntu5) 
```
- File and patch attacked

Jano

**Comment by aircrack-ng** [comment]
**Attachment by Jano on 3 Apr 2014 16:37 UTC**

Uploaded file [installation-ubuntu-32bit.txt](https://github.com/aircrack-ng/trac-attachments/blob/master/1299/20140403-163751_976-installation-ubuntu-32bit.txt)

installation without display "setuid" error on Ubuntu 32-bit

**Comment by aircrack-ng** [comment]
**Attachment by Jano on 3 Apr 2014 17:01 UTC**

Uploaded file [setuid_airbase_tkiptun.patch](https://github.com/aircrack-ng/trac-attachments/blob/master/1299/20140403-170148_455-setuid_airbase_tkiptun.patch)

Add display error when setuid fails to Airbase-ng and Tkiptun-ng

**Comment by aircrack-ng** [comment]
**Comment by misterx on 3 Apr 2014 16:35 UTC**

There seems to be something missing for airbase-ng patch, you simply remove the setuid()

**Comment by aircrack-ng** [comment]
**Comment by misterx on 3 Apr 2014 16:35 UTC**

Fixed in aircrack-ng/aircrack-ng@f60291bcca99dba82a0cb6f4a8490f1c791132dc.
